crates/ty_python_semantic/src/semantic_index/use_def/place_state.rs

```diff
@@ -176,6 +176,15 @@ impl Declarations {
             }
         }
     }
+
+    pub(super) fn mark_reachability_constraints(
```

Comment:
Nit: We might want to call this `finalize` or `finish` so that we also can call `shrink_to_fit` on the `live_declarations`